File: 338-CountingBits.js
// Given an integer n, return an array ans of length n + 1 such that for each i (0 <= i <= n), ans[i] is the number of 1's in the binary representation of i.
// Input: n = 2
// Output: [0,1,1]
// Explanation:
// 0 --> 0
// 1 --> 1
// 2 --> 10

function convertBinary(n) {
    let sum = 0;
    while (n > 0) {
        if (n % 2 === 1) sum += 1;
        n = Math.floor(n / 2);
    }
    return sum
}

function countBits(n) {
    let res = []; 
    for (let i = 0; i <= n; i++) {
        res.push(convertBinary(i));
    }
    return res
}
console.log(countBits(2))

File: 338-CountingBits.py
def countBits(n):
    res = []
    for i in range(n + 1): 
        res.append(0)
    for i in range(1,len(res)):
        res[i] = res[i >> 1] + (i & 1)
    return res
print(countBits(5))
